core/database/db.py

```diff
@@ -90,6 +109,35 @@ def get_entire_partition(pk):
     return result
 
 
+def scan_sk_for_substr(substr):
```

Comment:
Its a shame that dynamodb syntax is so ugly, but as long as this works when you tried it I'll take your word for it!